### PR TITLE
Layout as S3

### PR DESCRIPTION
**Pull request by JohnCoene**
This implements the `layout` callback as an S3 method.

Also, given the improvements made for #236 and #237 I've removed all custom methods. These methods were achieving what the default now does (layout by rows and columns) and are thus no longer needed.

### CODE REVIEW

File: R/layout.R [hunk 1]
```diff
@@ -47,166 +88,3 @@ layout_fields <- function(fields) {
...
-    ),
-    div(
-      id = id,
-      class = "collapse",
-      fields_hidden |> layout_fields()
-    )
-  )
-}
-
-plot_layout_fields <- function(fields, ...) {
```
Comment (nbenn) [inline R/layout.R:85]: Do we not need this as `layout.plot_block ()` (and the following accordingly)? Please disregard my comment if the deletion (also for filter, summarize, etc.) is on purpose.

**Review by nbenn — APPROVED** [review]
Lgtm, thanks!

File: R/layout.R [hunk 2]
```diff
@@ -1,4 +1,45 @@
...
+#' @param x Object (block).
+#' @param fields Named list of fields.
+#' @param ... Currently ignored.
+#' @export
+layout <- function(x, fields, ...) UseMethod("layout")
+
+#' @export
+layout.block <- function(x, fields, ...) {
+  # we remove hidden fields
+  fields <- fields[lengths(fields) > 0L]
```
[the remark this answers is not in the document]
Comment (JohnCoene) [reply R/layout.R:13]: Happy to change that, any suggestions?

I added that line when the default layout was redone, it's just to omit the hidden fields from the UI.

[the remark this answers is not in the document]
Comment (nbenn) [reply R/layout.R:13]: I was thinking about checking inheritance, i.e. `lgl_ply(fields, Negate(inherits), "hidden_field")`, or so, where `lgl_ply()` is an internal wrapper for `vapply()`.

File: R/layout.R (hunk 1 above)
Comment (nbenn) [reply R/layout.R:85]: Ah, sorry, didn't pay close enough attention. Even better!

File: R/layout.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohnCoene) [reply R/layout.R]: Did I get this right?

https://github.com/blockr-org/blockr/pull/251/commits/fb5ded0c18155a63c0fe29cb9498e8451d7d2093

**Review by DivadNojnarg — APPROVED** [review]
Thanks!

**Pull request merged**